packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts

```diff
@@ -607,10 +692,169 @@ export abstract class BaseService extends Resource
         this.executeCommandLogConfiguration();
       }
     }
+    this.node.addValidation({
+      validate: this.validateDeploymentAlarms.bind(this),
+    });
     this.node.defaultChild = this.resource;
   }
 
-  /**   * Enable Service Connect
+  /**
+   * Enable Deployment Alarms which take advantage of arbitrary alarms and configure them after service initialization.
+   * If you have already enabled deployment alarms, this function can be used to tell ECS about additional alarms that
+   * should interrupt a deployment.
+   *
+   * New alarms specified in subsequent calls of this function will be appended to the existing list of alarms.
+   *
+   * Subsequent calls of this function will respect the Alarm Behavior you set previously, unless you specify a new
+   * `behavior`.
```

Comment:
[p0] This goes against the design principles of CDK constructs. While you use imperative languages to write your CDK apps, the resources are defined in a declarative way. The order in which you call functions should not impact the resulting infrastructure. Instead, we will just throw an error if the user tries to mix different Alarm Behaviors. What is the intent behind that anyway? Why should the last specified behavior be the one that "wins"?